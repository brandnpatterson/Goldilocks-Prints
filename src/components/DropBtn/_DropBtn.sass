@import ../../index

.drop-btn
  cursor: pointer
  display: none
  +mediaQuery($small)
    display: inline-block
  right: 20px
  position: fixed
  z-index: 10
  div
    width: 35px
    background: $black
    border-radius: 25px
    height: 5px
    margin: 6px 0
.bar1,
.bar2,
.bar3
  width: 35px
  height: 5px
  margin: 6px 0
.transform
  .bar1
    transform: rotate(-45deg) translate(-9px, 6px)
  .bar2
    opacity: 0
  .bar3
    transform: rotate(45deg) translate(-9px, -6px)

.nav-links-responsive
  .responsive-divider-one,
  .responsive-divider-two
    display: block
    background: $black
    top: 0
    height: 100vh
    width: 5px
    position: absolute
    z-index: 2
  .responsive-divider-one
    transform: translate(130px, -42px) rotate(20deg)
  .responsive-divider-two
    transform: translate(240px, -42px) rotate(-20deg)
