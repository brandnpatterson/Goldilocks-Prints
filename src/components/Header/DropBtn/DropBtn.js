import React, { Component } from 'react'
import Links from '../../Links/Links'
import './DropBtn.sass'

export default class DropBtn extends Component {
  constructor () {
    super()
    this.state = {
      transform: false
    }
    this.onDropBtnSelect = this.onDropBtnSelect.bind(this)
    this.onNavSelect = this.onNavSelect.bind(this)
  }
  onDropBtnSelect () {
    if (this.state.transform === false) {
      this.setState({
        transform: true
      })
    } else if (this.state.transform === true) {
      this.setState({
        transform: false
      })
    }
  }
  onNavSelect () {
    if (this.state.transform === true) {
      this.setState({
        transform: false
      })
    }
  }
  render () {
    return (
      <div>
        <div
          className={'drop-btn ' + (this.state.transform ? 'transform' : '')}
          onClick={this.onDropBtnSelect}
        >
          <div className='bar1' />
          <div className='bar2' />
          <div className='bar3' />
        </div>
        <div
          className={'nav-links' + (this.state.transform ? '-responsive' : '')}
          onClick={this.onNavSelect}
        >
          <Links />
        </div>
      </div>
    )
  }
}
