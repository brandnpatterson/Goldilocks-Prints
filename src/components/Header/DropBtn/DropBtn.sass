@import ../../../App.sass
/* drop-btn */
.drop-btn
  cursor: pointer
  display: none
  +mediaQuery($small)
    display: inline-block
  right: 20px
  position: fixed
  z-index: 10
  div
    width: 35px
    background: $dark
    border-radius: 25px
    height: 5px
    margin: 6px 0
.bar1,
.bar2,
.bar3
  width: 35px
  height: 5px
  margin: 6px 0
  transition: all 0.4s ease-out
.transform
  .bar1
    transform: rotate(-45deg) translate(-9px, 6px)
  .bar2
    opacity: 0
  .bar3
    transform: rotate(45deg) translate(-9px, -6px)
