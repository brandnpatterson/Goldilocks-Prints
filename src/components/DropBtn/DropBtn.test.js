import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import DropBtn from './DropBtn'

test('It should render the DropBtn component as expected', () => {
  const component = shallow(<DropBtn />)
  const tree = toJson(component)
  expect(tree).toMatchSnapshot()
})
