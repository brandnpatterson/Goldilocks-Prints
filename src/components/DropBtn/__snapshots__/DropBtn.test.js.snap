// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render the DropBtn component as expected 1`] = `
<div>
  <div
    className="drop-btn "
    onClick={[Function]}
  >
    <div
      className="bar1"
    />
    <div
      className="bar2"
    />
    <div
      className="bar3"
    />
  </div>
  <div
    className="nav-links"
    onClick={[Function]}
  >
    <Links />
  </div>
</div>
`;
